### ISSUE TITLE
Ansible hostvars[] directory seems to assume different path for host_vars with --limit is used

### ISSUE THREAD

**Issue by feliksik** [ansible/ansible#12174]
This bug report is a result investigating https://groups.google.com/forum/#!searchin/ansible-project/limit$20hostvars$20/ansible-project/ZZ12KPiOE2s/rfdotpnmKgAJ . 

It turns out that the host_vars/ directory must be relative to the inventory file _when the host is not included in --limit_. Ansible can define any such requirement, but note that it is inconsistent to be more liberal with the placement of the host_vars directory when a file is not excluded with --limit (i.e., if the host is included in the run, either the directory relative to the playbook can be used, or the directory relative to the inventory file). 

To reproduce, see the project in http://feliksik.nl/ansible-limit.tgz structure, which structure looks as follows (modify inventory to your environment, of course). 

```
.
├── host_vars
│   └── outlimit
├── inventory
├── playbook.yml
└── sub
    └── inventory -> ../inventory
```

There is the following calls possible: 
root1) ansible-playbook -i inventory playbook.yml --limit inlimit
root2) ansible-playbook -i inventory playbook.yml --limit inlimit,outlimit
sub1) ansible-playbook -i sub/inventory playbook.yml --limit inlimit
sub2) ansible-playbook -i sub/inventory playbook.yml --limit inlimit,outlimit

Expected behaviour: 
- if root2 works, then root1 should also work
- if sub2 works (i.e. it is legal for the host_vars directory to not be in the same directory as the inventory file), then sub1 should also work.

Actual behavior: 
- root1 works
- root2 works
- sub1 **fails**! 
- sub2 works

**Comment by jimi-c** [comment]
@feliksik with which version of Ansible are you seeing this behavior? If you're not running against devel already, could you also run this against that to see if the issue persists or if there are any other differences? The handling of paths have been somewhat cleaned up in the devel branch (2.0) so we want to make sure the problem hasn't already been addressed. Thanks!

**Comment by feliksik** [comment]
This is 1.9.2.
@jimi-c I'd be happy to help investigate that, but it'll take me a few days to get to this.

**Comment by jimi-c** [comment]
@feliksik no problem, just let us know.

Thanks!

**Comment by feliksik** [comment]
I will not be able to work on this until October, as holiday kicks in. 
The only thing is that I need to install a dev version of Ansible. If anyone has it installed, it should cost you a few minutes to set up the ip address in your inventory and run the script as provided on http://feliksik.nl/ansible-limit.tgz

**Comment by feliksik** [comment]
Thank for you patience. I have checked, and it turns out the bug does not exist in current 2.0 dev. 

That means the problem exists only in the current 1.9 version.

**Comment by abadger** [comment]
This should be fixed by #15130.  So this should now be fixed in both devel and stable-1.9 branches.
